src/coreclr/jit/hwintrinsiccodegenarm64.cpp

```diff
@@ -796,15 +796,127 @@ void CodeGen::genHWIntrinsic(GenTreeHWIntrinsic* node)
+                {
+                    // op1 of TYP_SIMD12 should be considered as TYP_SIMD16
+                    simdType = TYP_SIMD16;
+                }
+
+                if (!intrin.op2->OperIsConst())
+                {
+                    assert(!intrin.op2->isContained());
+
+                    emitAttr baseTypeSize  = emitTypeSize(intrin.baseType);
+                    unsigned baseTypeScale = genLog2(EA_SIZE_IN_BYTES(baseTypeSize));
+
+                    regNumber baseReg;
+                    regNumber indexReg = op2Reg;
+
+                    // Optimize the case of op1 is in memory and trying to access ith element.
+                    if (!intrin.op1->isUsedFromReg())
+                    {
+                        assert(intrin.op1->isContained());
+
+                        if (intrin.op1->OperIsLocal())
+                        {
+                            unsigned varNum = intrin.op1->AsLclVarCommon()->GetLclNum();
+                            baseReg         = node->ExtractTempReg();
+
+                            // Load the address of varNum
+                            GetEmitter()->emitIns_R_S(INS_lea, EA_PTRSIZE, baseReg, varNum, 0);
+                        }
+                        else
+                        {
+                            // Require GT_IND addr to be not contained.
+                            assert(intrin.op1->OperIs(GT_IND));
+
+                            GenTree* addr = intrin.op1->AsIndir()->Addr();
+                            assert(!addr->isContained());
+                            baseReg = addr->GetRegNum();
+                        }
+                    }
+                    else
+                    {
+                        unsigned simdInitTempVarNum = compiler->lvaSIMDInitTempVarNum;
+                        noway_assert(simdInitTempVarNum != BAD_VAR_NUM);
+
+                        baseReg = node->ExtractTempReg();
+
+                        // Load the address of simdInitTempVarNum
+                        GetEmitter()->emitIns_R_S(INS_lea, EA_PTRSIZE, baseReg, simdInitTempVarNum, 0);
+
+                        // Store the vector to simdInitTempVarNum
+                        GetEmitter()->emitIns_R_R(INS_str, emitTypeSize(simdType), op1Reg, baseReg);
+                    }
+
+                    assert(genIsValidIntReg(indexReg));
+                    assert(genIsValidIntReg(baseReg));
+                    assert(baseReg != indexReg);
+
+                    // Load item at baseReg[index]
+                    GetEmitter()->emitIns_R_R_R_Ext(ins_Load(intrin.baseType), baseTypeSize, targetReg, baseReg,
+                                                    indexReg, INS_OPTS_LSL, baseTypeScale);
+                }
+                else if (!GetEmitter()->isValidVectorIndex(emitTypeSize(simdType), emitTypeSize(intrin.baseType),
+                                                           intrin.op2->AsIntCon()->IconValue()))
+                {
```

Comment:
The runtime will have generated a bounds check and it will fail if the index is invalid.

Ideally we'd just do what x64 does and generate a masked load here, but I preserved what a64 was already doing to simplify the change/diffs